src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs

```diff
@@ -358,7 +358,9 @@ private SyntaxToken RenameAndAnnotate(SyntaxToken token, SyntaxToken newToken, b
                         symbol = symbol.ContainingSymbol;
                     }
 
-                    var sourceDefinition = SymbolFinder.FindSourceDefinition(symbol, _solution, _cancellationToken);
+                    // We cannot make this containing method async since it's being used in a rewriter. FindSourceDefinitionAsync will only yield in cross-language cases
+                    // when the compilation is not already available, so this is expected to not really cause any significant blocking.
+                    var sourceDefinition = SymbolFinder.FindSourceDefinitionAsync(symbol, _solution, _cancellationToken).WaitAndGetResult_CanCallOnBackground(_cancellationToken);
```

Comment:
This doesn't make me happy having this here, but it's a cross-language case to start with. If we're concerned I can probably refactor this a bit. Looking at the code, the symbol here only is checked for:

1. Being a namespace symbol, which doesn't matter that we have the source symbol.
2. Being a named type symbol that's a delegate and has an associated symbol.

In the 2 case we could at least do the delegate type check first, and only then hop to the VB symbol to see if it's actually implicitly defined.

Reply:
this is fine for me.